### PR TITLE
#4851 ConcurrentHashSet→Utils.concurrentHashSet

### PR DESCRIPTION
**Pull request by magicprinc**
Issue: https://github.com/eclipse-vertx/vert.x/issues/4851

It seems that io.vertx.core.impl.ConcurrentHashSet is a copy of JDK's Collections.newSetFromMap.
The Best Code is No Code At All.
It is probably a good idea to review and replace this class with two utility methods.

### CODE REVIEW

File: src/main/java/io/vertx/core/impl/Utils.java [hunk 1]
```diff
@@ -45,4 +49,11 @@ public static boolean isWindows() {
     return isWindows;
   }
 
+  public static <E> Set<E> concurrentHashSet (){
```
Comment (tsegismont) [inline src/main/java/io/vertx/core/impl/Utils.java:52]: It is not necessary to add new methods.

Where a `ConcurrentHashSet` is needed, we can use `java.util.concurrent.ConcurrentHashMap#newKeySet()`

File: src/test/java/io/vertx/core/ComplexHATest.java [hunk 2]
```diff
@@ -40,11 +37,11 @@
  */
 public class ComplexHATest extends VertxTestBase {
 
-  protected ClusterManager getClusterManager() {
+  @Override protected ClusterManager getClusterManager() {
```
Comment (tsegismont) [inline src/test/java/io/vertx/core/ComplexHATest.java:40]: Please configure the formatter of your IDE or text editor. The annotation should be on a separate line above.

File: src/test/java/io/vertx/core/eventbus/LocalEventBusTest.java [hunk 3]
```diff
@@ -41,7 +41,7 @@ public class LocalEventBusTest extends EventBusTestBase {
   private EventBusInternal eb;
   private boolean running;
 
-  public void setUp() throws Exception {
+  @Override public void setUp() throws Exception {
```
Comment (tsegismont) [inline src/test/java/io/vertx/core/eventbus/LocalEventBusTest.java:44]: Please configure the formatter of your IDE or text editor. The annotation should be on a separate line above.

File: src/test/java/io/vertx/core/eventbus/LocalEventBusTest.java [hunk 4]
```diff
@@ -52,7 +52,7 @@ public void setUp() throws Exception {
     running = true;
   }
 
-  protected void tearDown() throws Exception {
+  @Override protected void tearDown() throws Exception {
```
Comment (tsegismont) [inline src/test/java/io/vertx/core/eventbus/LocalEventBusTest.java:55]: Please configure the formatter of your IDE or text editor. The annotation should be on a separate line above.

**Review by tsegismont — CHANGES_REQUESTED** [review]
Thank you @magicprinc 

Can you please make a few changes?

File: src/main/java/io/vertx/core/impl/Utils.java (hunk 1 above)
Comment (He-Pin) [reply src/main/java/io/vertx/core/impl/Utils.java:52]: Yes, use this method, I used it in my pekko pr too.

**Comment by magicprinc** [comment]
@tsegismont fixed. Please see.

Changed implementation to 
```java
public static <E> Set<E> concurrentHashSet (){
  return ConcurrentHashMap.newKeySet();// we can change implementation
}
```
or do you mean replace Util.concurrentHashSet with its body.

I think this separate method is a more robust and readable solution.

**Comment by magicprinc** [comment]
@He-Pin Could you fix other Vert.x modules (if any) which are using this - deleted - class

**Comment by He-Pin** [comment]
@magicprinc Hi, thanks for the ping, I think it would be better for you to do it, and I don't have a write access to this project, then I have to make a PR to yours and that will make the thing a little complicated :)

**Comment by magicprinc** [comment]
@He-Pin @tsegismont I can't find any other ConcurrentHashSet usages besides the corrected ones, so we probably have managed it :-)

File: src/main/java/io/vertx/core/impl/Utils.java (hunk 1 above)
Comment (tsegismont) [reply src/main/java/io/vertx/core/impl/Utils.java:52]: Please remove this method, it is not necessary.

**Comment by tsegismont** [comment]
@magicprinc before we merge this PR we also need updates for the projects that depend on this class (see https://github.com/eclipse-vertx/vert.x/issues/4851#issuecomment-1713554985)

**Comment by magicprinc** [comment]
@tsegismont All your tasks are done! 👍
1) I have replaced method with its body.
2) Fix in vertx-service-discovery: https://github.com/vert-x3/vertx-service-discovery/pull/172
3) Fix in vertx-mqtt: https://github.com/vert-x3/vertx-mqtt/pull/243

File: src/main/java/io/vertx/core/impl/DeploymentManager.java [hunk 5]
```diff
@@ -122,24 +121,24 @@ public Future<Void> undeployAll() {
...
-    context.runOnContext(v -> {
-      try {
-        completionHandler.handle(result);
-      } catch (Throwable t) {
-        log.error("Failure in calling handler", t);
-        throw t;
-      }
-    });
-  }
+//  private <T> void reportFailure(Throwable t, ContextInternal context, Promise<T> completionHandler) {
```
Comment (tsegismont) [inline src/main/java/io/vertx/core/impl/DeploymentManager.java:124]: Can you please remove the methods instead of commenting?

**Review by tsegismont — COMMENTED** [review]
@magicprinc one last comment, thank you

File: src/main/java/io/vertx/core/impl/DeploymentManager.java (hunk 5 above)
Comment (magicprinc) [reply src/main/java/io/vertx/core/impl/DeploymentManager.java]: Deleted!

Comment (tsegismont) [reply src/main/java/io/vertx/core/impl/DeploymentManager.java]: Thank you

**Review by tsegismont — APPROVED** [review]

**Comment by magicprinc** [comment]
@He-Pin Could you review this too? 🙏

**Review by He-Pin — COMMENTED** [review]
Lgtm

**Pull request merged**